workers/worker_scheduler.rb

```diff
@@ -25,11 +32,25 @@ def initialize(sleep_interval: nil, cron_schedule: nil)
     end
 
     def schedule(&block)
-      if !self.sleep_interval.nil?
+      if self.sleep_interval
         block.call
         Kernel.sleep(self.sleep_interval)
-      elsif !self.cron_schedule.nil?
-        self.scheduler.cron(self.cron_schedule) { block.call }
+      elsif self.cron_schedule && self.scheduled_cron_job.nil?
+        job_id = self.scheduler.cron(self.cron_schedule) do
+          self.scheduled_cron_job = nil
+          block.call
+        end
+
+        self.scheduled_cron_job = self.scheduler.job(job_id)
+        logger.debug("Scheduling cron job for #{self.cron_schedule}.")
+        logger.debug("Next time #{self.scheduled_cron_job.next_time} or #{(self.scheduled_cron_job.next_time - Time.now) / (60 * 60)} hours from now.")
+      end
+    end
+
+    # Shuts down the scheduler, ceases any scheduler/triggering activity.
+    def shutdown
+      if self.scheduler
+        self.scheduler.shutdown
```

Comment:
can be rewritten as `self.scheduler.shutdown if self.scheduler` ❤️ ruby for this